docs/reusing-typechecking-results.md

```diff
@@ -0,0 +1,171 @@
+|Encoding OptData                    |  0,8899|  0,0024|    173|      0|      0|      0|    378|     30|
+|TailCall Checks                     |  0,8990|  0,0025|    173|      0|      0|      0|    378|     30|
+// main4, main5
+|TAST -> IL                          |  0,9487|  0,0447|    176|      0|      0|      0|    378|     30|
+// main6
+|Write .NET Binary                   |  1,1530|  0,1972|    180|      0|      0|      0|    384|     30|     <-- long
+--------------------------------------------------------------------------------------------------------
+```
+
+**CE100xnest5.fs**
+```
+--------------------------------------------------------------------------------------------------------
+|Phase name                          |Elapsed |Duration| WS(MB)|  GC0  |  GC1  |  GC2  |Handles|Threads|
+|------------------------------------|--------|--------|-------|-------|-------|-------|-------|-------|
+// main1
+|Import mscorlib+FSharp.Core         |  0,4092|  0,4084|    101|      0|      0|      0|    365|     30|     <-- long
+|Parse inputs                        |  0,4635|  0,0445|    108|      0|      0|      0|    374|     30|
+|Import non-system references        |  0,5475|  0,0775|    128|      0|      0|      0|    374|     30|
+|Typecheck                           |  3,1157|  2,5612|    198|     18|     15|      3|    712|     45|     <-- long
+|Typechecked                         |  3,1219|  0,0002|    198|      0|      0|      0|    712|     45|
+// main2
+|Write Interface File                |  3,1280|  0,0000|    198|      0|      0|      0|    712|     45|
+|Write XML doc signatures            |  3,1363|  0,0000|    198|      0|      0|      0|    712|     45|
+|Write XML docs                      |  3,1435|  0,0002|    198|      0|      0|      0|    712|     45|
+// main3
+|Encode Interface Data               |  3,1803|  0,0296|    198|      0|      0|      0|    712|     45|
+|Optimizations                       |  8,9949|  5,8049|    216|     43|     42|      2|    457|     45|     <-- longest
+|Ending Optimizations                |  9,0015|  0,0000|    216|      0|      0|      0|    457|     45|
+|Encoding OptData                    |  9,0090|  0,0010|    216|      0|      0|      0|    457|     45|
+|TailCall Checks                     |  9,0190|  0,0013|    216|      0|      0|      0|    457|     45|
+// main4, main5
+|TAST -> IL                          |  9,0463|  0,0210|    217|      0|      0|      0|    458|     46|
+// main6
+|Write .NET Binary                   |  9,2463|  0,1930|    218|      0|      0|      0|    464|     46|     <-- long
+--------------------------------------------------------------------------------------------------------
+```
+
+The phases taking the longest are marked in the right. Those are assembly import, type check, optimizations and IL writing.
+
+Optimizations are not relevant in the dev loop with run/debug/test cycles so we won't take those into account.
+
+## Implementation plan
```

Comment:
Can you also please add any potential challenges in implementing each stage?

Reply:
Apart from the whole novelty approach in the stage 2, I cannot come up with anything remarkable. I can imagine edge cases everywhere but that's expected here I guess.